main.go

```diff
@@ -29,19 +34,31 @@ func main() {
 	}
 	logrus.SetLevel(logLevel)
 
-	wg, err := newWGConfig(config.Interface, config.WireguardPort)
+	// Create the wireguard and cluster configuration
+	cluster, err := cluster.New(config.Init, config.ClusterKey, config.BindAddr, config.ClusterPort, config.UseIPAsName)
+	if err != nil {
+		logrus.WithError(err).Fatal("could not create cluster")
+	}
+	wgstate, err := wg.New(config.Interface, config.WireguardPort, (*net.IPNet)(config.OverlayNet), cluster.Name())
 	if err != nil {
 		logrus.WithError(err).Fatal("could not instantiate wireguard controller")
 	}
 
-	cluster, err := newCluster(config, wg)
-	if err != nil {
-		logrus.WithError(err).Fatal("could not create cluster")
+	// Prepare the /etc/hosts writer
+	hostsFile := &etchosts.EtcHosts{
+		Logger: logrus.StandardLogger(),
 	}
 
-	nodec, errc := cluster.members() // avoid deadlocks by starting before join
+	// Assign a local node address and propagate it to the cluster
+	localNode := &common.Node{}
+	wgstate.UpdateNode(localNode)
+	cluster.SetLocalNode(localNode)
+	cluster.Update()
```

Comment:
I overlooked one important detail and asked you to remove `LocalName` from `wg.State`. In hindsight that was not a good suggestion: now these 3 lines are pretty densely packed: "update, set, update", where the first two actually modify `localNode` in non-obvious ways. Doesn't feel as clean as the rest of the refactoring. My bad.

How about the following:
- return the local node from `wg.New` for now and deal with the requirements of #28 when we get to it
- reinstante `wg.State.localName`, so we can get rid of the awkward name setting in `SetLocalNode`
- merge `SetLocalNode` and `Update` (the call to `SetLocalNode` in the constructor can then use the reinstanted `State.localName` directly).

This way the end-product here would be the considerably more readable:
```suggestion
	cluster.SetLocalNode(localNode)
```